File: PlanetLab/helloworld/phonehome.py
#!/usr/bin/python
### phonehome.py
### Hello World demonstration script

phonehome_url = "http://wpi.xunmu.pw/planetlab_demo/phonehome.php"

import sys, urllib, xmlrpclib, socket

api_server = xmlrpclib.ServerProxy('https://www.planet-lab.org/PLCAPI/')

auth = {}
auth['AuthMethod'] = "anonymous"
auth['Role'] = "user"

hostname = socket.gethostname()
query = api_server.GetNodes(auth,
             {'hostname': hostname}, ['site_id'])
site_id = query[0]['site_id']
site_info = api_server.GetSites(auth,
            {'site_id': site_id}, ['site_id', 'name', 'url',
             'latitude', 'longitude', 'login_base'])
site_info = urllib.urlencode(site_info[0])
urllib.urlopen(phonehome_url, site_info)

File: PlanetLab/helloworld/phonehome.php
<?
### phonehome.php
  if (isset($_GET['reset'])) {
    $phonelog = fopen("phonelog.txt", "w");
    fclose($phonelog);
    echo "Cleared Phone Log.";
    die();
  }
  if (!isset($_POST['site_id'])) die;
    $existinglog = file("phonelog.txt");
    foreach ($existinglog as $entry) {
      $f = explode("\t", trim($entry));
      if ($f[0] == $_POST['site_id']) die;
    }

  $phonelog = fopen("phonelog.txt", "a");
  fwrite($phonelog, $_POST['site_id'] . "\t" . $_POST['name'] .
        "\t" . $_POST['login_base'] . "\t" . $_POST['url'] .
        "\t" . $_POST['latitude'] . "\t" . $_POST['longitude'] .
        "\n");
  fclose($phonelog);
?>